### PR TITLE
feat: change datepicker to have better ui and better functionality

### PR DESCRIPTION
**Pull request by ParhamTehrani**
## feat: change datepicker to have better ui and better functionality

- **Base/Head**: `release-5.4` ← `release-5.4-feat-datepicker`
- **Scope**: 10 files changed, +1,904 / −73
- **PR**: https://github.com/wp-slimstat/wp-slimstat/pull/100

### Summary
Replace the existing datepicker with a modern, accessible, and more capable component to improve Slimstat’s date/range filtering UX within WP Admin.

### Motivation
- Current picker is confusing for ranges and not optimized for mobile.
- Reduce clicks for common timeframes and make active selections clearer.

### Key Changes
- New datepicker with:
  - Quick presets: Today, Yesterday, Last 7/30 days, This/Last Month, Custom Range
  - Clear start/end range visualization and keyboard navigation
  - Improved i18n/locale support and date formatting consistency
- UI aligned with WordPress admin styles; responsive and touch-friendly
- Refined date parse/format utilities and validation for invalid ranges

### Accessibility
- Better focus states and ARIA attributes
- Logical tab order and higher-contrast states

### Backward Compatibility
- Preserves existing date filter behavior and query params
- Accepts legacy input formats; normalizes internally
- No breaking API changes expected

### Performance
- Lazy initializes the picker; loads only required assets

### Security/Privacy
- No PII collected; client-side handling
- Sanitizes inputs before use in queries

### QA Checklist
- Verify all presets and custom ranges (including month/year boundaries)
- Confirm keyboard navigation and screen reader announcements
- Test multiple locales/formatting and RTL
- Ensure selected range persists in URL/state and reports reflect it

### Notes
- Targets `release-5.4`
- Consider adding short docs in README/admin help about presets and custom ranges

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1211123781037812

---

> [!NOTE]
> Replace the legacy date filters with a modern date range picker wired to URL presets/from-to and server-side helpers, updating filters, assets, and defaults.
> 
> - **Admin UI/Frontend**:
>   - **New date range picker** in `admin/assets/js/daterangepicker/slimstat-daterangepicker.js` using `daterangepicker`/`moment` with presets, custom range, timezone handling, keyboard/ARIA, URL sync (`type`, `from`, `to`), and cache-clear hook.
>   - **Assets**: enqueue `moment`, `daterangepicker` JS/CSS and custom styles; localize options/strings via `SlimStatDatePicker` in `admin/index.php`.
>   - **View**: replace legacy date inputs/dropdowns with a compact picker button and hidden input in `admin/view/index.php`.
> - **Backend/PHP**:
>   - **New `DateRangeHelper`**: WP timezone/week-start/date-format mapping, preset ranges, current range detection, validation, and display formatting.
>   - **Filtering**: `wp_slimstat_db` parses `type` presets and `from`/`to` to `strtotime`/`interval`; switches several date ops to `wp_date`; disables query caching; minor range calc tweak.
> - **Defaults**:
>   - Change `posts_column_day_interval` fallback from `30` to `28`.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 39df99af1fc13f1adbe5ef4a503ce86db6b71946. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: src/Components/DateRangeHelper.php
Comment (mostafasoufi) [inline src/Components/DateRangeHelper.php:1]: If it's a new class/file, let's move it under new structure `/src` and camelCase naming, please.

File: admin/view/index.php [hunk 1]
```diff
@@ -44,79 +44,42 @@
...
-if (
-    wp_slimstat_db::$filters_normalized['date']['day'] != intval(date_i18n('j')) || wp_slimstat_db::$filters_normalized['date']['month'] != intval(date_i18n('n')) || wp_slimstat_db::$filters_normalized['date']['year'] != intval(date_i18n('Y')) || (wp_slimstat_db::$filters_normalized['date']['interval'] != -abs(wp_slimstat::$settings['posts_column_day_interval']) && wp_slimstat_db::$filters_normalized['date']['interval'] != -intval(date_i18n('j')) + 1)
-) {
-    echo '<a class="slimstat-filter-link button button-secondary noslimstat" data-reset-filters="true" href="' . wp_slimstat_reports::fs_url() . '">' . __('Reset Filters', 'wp-slimstat') . '</a>';
-}
-?>
+                <?php
+                // Include the date range helper for display logic
+                if (class_exists('SlimStat_DateRange_Helper') === false) {
+                    include_once(__DIR__ . '/class-slimstat-daterange-helper.php');
```
Comment (mostafasoufi) [inline admin/view/index.php]: Let's use the component `View:class`

File: admin/index.php [hunk 2]
```diff
@@ -707,6 +707,49 @@ public static function wp_slimstat_enqueue_scripts($_hook = '')
...
+            if (str_contains($page, 'slim') && !str_contains($page, 'setting')) {
+                $should_load_datepicker = true;
+            }
+        }
+        
+        if ($should_load_datepicker) {
+            
+            // Include the date range helper
+            if (class_exists('SlimStat_DateRange_Helper') === false) {
+                include_once(__DIR__ . '/view/class-slimstat-daterange-helper.php');
```
Comment (mostafasoufi) [inline admin/index.php]: Same here.

**Review by mostafasoufi — CHANGES_REQUESTED** [review]

**Comment by parhumm** [comment]
@Cursor review

File: admin/index.php [hunk 3]
```diff
@@ -707,6 +707,44 @@ public static function wp_slimstat_enqueue_scripts($_hook = '')
...
+            // Enqueue date picker styles
+            wp_enqueue_style('slimstat_daterangepicker_base', plugins_url('/admin/assets/css/daterangepicker/daterangepicker.css', __DIR__), [], '3.1.0');
+            wp_enqueue_style('slimstat_daterangepicker_custom', plugins_url('/admin/assets/css/daterangepicker/slimstat-datepicker-styles.css', __DIR__), ['slimstat_daterangepicker_base'], SLIMSTAT_ANALYTICS_VERSION);
+            
+            // Localize date picker script
+            $datepicker_params = [
+                'ajax_url' => admin_url('admin-ajax.php'),
+                'clear_cache_nonce' => wp_create_nonce('slimstat_clear_cache'),
+                'options' => [
+                    'wp_timezone' => \SlimStat\Components\DateRangeHelper::get_wp_timezone(),
```
Comment (fstarlike) [inline admin/index.php]: Where classes from other namespaces are going to be used, they should be imported with use before being used in the code.

**Review by fstarlike — COMMENTED** [review]
I reviewed it and it looks fine.

**Comment by parhumm** [comment]
@cursor review

**Comment by parhumm** [comment]
@cursor review

**Comment by parhumm** [comment]
@Cursor review

**Comment by parhumm** [comment]
@cursor review
